target_disk_darwin.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:build darwin
+// +build darwin
+
+package extract
+
+import (
+	"time"
+
+	"golang.org/x/sys/unix"
+)
+
+// Lchtimes modifies the access and modified timestamps on a target path
+// This capability is only available on unix as of now.
+func Lchtimes(path string, atime, mtime time.Time) error {
+
+	return unix.Lutimes(path, []unix.Timeval{
+		{Sec: atime.Unix(), Usec: int32(atime.Nanosecond() / 1e6 % 1e6)},
+		{Sec: mtime.Unix(), Usec: int32(mtime.Nanosecond() / 1e6 % 1e6)},
```

Comment:
These calculations have me confused... The `time.Time.Nanosecond` call gives you the nanosecond offset within the second, so you shouldn't need to take the modulo... Also to go from nano `1e9` to Micro `1e6` you need to divide by 1000 `1e3`. I can't see a test covering this, so it might be a good idea to encapsulate the conversion somewhere and wrap it in a test, and maybe you'd end up with something like this?

The same comment applies to other instances below.

```suggestion
		{Sec: atime.Unix(), Usec: int32(atime.Nanosecond() / 1e3)},
		{Sec: mtime.Unix(), Usec: int32(mtime.Nanosecond() / 1e3)},
```

Reply:
Thanks for raising that point. I will investigate that part and prepare some tests for that 👌

Reply:
As a remark, the snippet to call the function was taken from go-slug ..
https://github.com/hashicorp/go-slug/blob/53c172a3e00fc1ac280365255b56af57846abdb4/internal/unpackinfo/lchtimes_darwin.go#L17-L18

Reply:
Hm, still looks odd there... There is also the [`time.Time.UnixMicro`](https://pkg.go.dev/time#Time.UnixMicro) func which you could use similarly, but that would need a `% 1e6`, e.g. `int32(atime.UnixMicro() % 1e6)` as it's absolute microseconds since the start of the epoch.

Reply:
Ooh I also just saw that both the `Sec` and `Usec` fields are `int64` so no need for the `int32` conversion!

Reply:
@samsalisbury you are 💯 the time conversion from nano to micro is by `1e3` and not `1e6`!

It depends on the local platform if `Sec` and `Usec` are `int64` or `int32`.

However, I adressed it in https://github.com/hashicorp/go-extract/pull/106/commits/4c99ad2e2ac810a4665550681f326939582c83bb and https://github.com/hashicorp/go-extract/pull/106/commits/fd3a38540c5cfd94b3b1feb83c051c0b3db92bc2

Reply:
re the reference to go-slug : https://github.com/hashicorp/go-slug/blob/53c172a3e00fc1ac280365255b56af57846abdb4/internal/unpackinfo/lchtimes_darwin.go#L17-L18

I think that must been a mistake

Reply:
This has been addressed by https://github.com/hashicorp/go-extract/pull/106/commits/593b990c2969df5fd3ccc84032036f3a30feab29